src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/ElasticSearch.php

```diff
@@ -85,4 +85,34 @@ public function getVersion(): string
 
         return $this->version;
     }
+
+    /**
+     * @return array
+     */
+    public function getTemplate(): array
+    {
+        $relationship = $this->environment->getRelationship('elasticsearch');
+
+        if (!$relationship) {
+            return [];
+        }
+
+        $esConfig = $relationship[0];
+
+        try {
+            $response = $this->clientFactory->create()->get(sprintf(
```

Comment:
This part is similar to in `getVersion` method, you can move it to a new private method to avoid duplication.
```
 $response = $this->clientFactory->create()->get(sprintf(
                    '%s:%s',
                    $esConfig['host'],
                    $esConfig['port']
                ));
                $esConfiguration = $response->getBody()->getContents();
                $esConfiguration = json_decode($esConfiguration, true);
```